pkg/component/runtime/manager_test.go

```diff
@@ -3512,14 +3522,14 @@ LOOP:
 			t.Logf("[subErrCh0] received: %v", err)
 			require.NoError(t, err)
 			count++
-			if count >= 2 {
+			if count >= 3 {
```

Comment:
This condition doesn't look reliable to me. I know the old one was wrong because the count _could_ go above 2 without actually hitting all expected states, but it's not obvious to me that both of them will _always_ send 3 "healthy" updates either. I'd feel better if test progress was based on reaching the target states (stopped for component 0 and healthy for component 1) rather than a magic number that depends on internal sequencing in the state reporting.